### PR TITLE
Migrate Knative-Eventing to Helm3

### PR DESCRIPTION
**Pull request by radufa**
**Description**

Changes proposed in this pull request:

- Migrate Knative-Eventing to Helm3
- Add "values.schema.json"

**Related issue(s)**

See #8524
See also #8517
Requires #8679

### CODE REVIEW

File: resources/knative-eventing/Chart.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: v1
+apiVersion: v2
 name: knative-eventing
 version: 0.12.0
 tillerVersion: ">=2.7.2-0"
```
Comment (nachtmaar) [inline resources/knative-eventing/Chart.yaml:4]: I guess tillerVersion is not required anymore, since there is no tiller in helm3

Comment (radufa) [reply resources/knative-eventing/Chart.yaml]: Done

File: resources/knative-eventing/charts/knative-eventing/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,6 @@
-apiVersion: v1
+apiVersion: v2
 name: knative-eventing
 version: 0.12.0
```
Comment (sayanh) [inline resources/knative-eventing/charts/knative-eventing/Chart.yaml:3]: version -> this is the Chart version
appVersion -> this is the version of the application
In this case, `0.12.0` should be the appVersion, IMO

File: resources/knative-eventing/Chart.yaml [hunk 3]
```diff
@@ -1,9 +1,11 @@
-apiVersion: v1
+apiVersion: v2
```
Comment (sayanh) [inline resources/knative-eventing/Chart.yaml:1]: `v1` would make the existing Kyma operator happy as well as helm3 client. Let's go with that for now. Once the complete migration is done, we will switch to v2.

File: resources/knative-eventing/Chart.yaml [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
 name: knative-eventing
 version: 0.12.0
-tillerVersion: ">=2.7.2-0"
 description: Helm chart for installing Knative Eventing
 keywords:
   - knative
   - knative-eventing
   - kyma
+dependencies:
```
Comment (sayanh) [inline resources/knative-eventing/Chart.yaml:9]: But then if we use `apiVersion: v1`, `dependencies` won't work hence we need to remove this.

Comment (radufa) [reply resources/knative-eventing/Chart.yaml:9]: For v1, we have to restore the "requirements.yaml" which defines the dependencies for v1. For V2, the dependencies should be moved into the chart, under the new "dependencies". This is an incompatible change in v2.

File: resources/knative-eventing/charts/knative-eventing/Chart.yaml (hunk 2 above)
Comment (radufa) [reply resources/knative-eventing/charts/knative-eventing/Chart.yaml:3]: I didn't change the version, it seems that it is the Knative Eventing version used.

File: resources/knative-eventing/Chart.yaml (hunk 3 above)
Comment (radufa) [reply resources/knative-eventing/Chart.yaml:1]: For this PR, it means that the "requirements.yaml" file which is needed only by v1 should be restored. This is an incompatible change in v2, see also my comment below.

**Pull request merged**